unity/Assets/Scripts/ProceduralTools.cs

```diff
@@ -2437,7 +2438,21 @@ CollisionDetectionMode collisionDetectionMode
                 meshObj.GetComponent<Renderer>().material = mat;
             }
 
-            mat.SetFloat("_Glossiness", 0f);
+            if (metallicSmoothnessTexturePath != null) {
+                metallicSmoothnessTexturePath = !Path.IsPathRooted(metallicSmoothnessTexturePath) ? Path.Combine(parentTexturesDir, metallicSmoothnessTexturePath) : metallicSmoothnessTexturePath;
+                 if (runtimePrefab == null) {
+                    runtimePrefab = go.AddComponent<RuntimePrefab>();
+                }
+                runtimePrefab.metallicSmoothnessTexturePath = metallicSmoothnessTexturePath;
+                mat.EnableKeyword("_METALLICGLOSSMAP");
+                byte[] imageBytes = File.ReadAllBytes(metallicSmoothnessTexturePath);
+                Texture2D tex = new Texture2D(2, 2);
+                tex.LoadImage(imageBytes);
+                
+                mat.SetTexture("_MetallicGlossMap", tex);
+            } else {
+                mat.SetFloat("_Glossiness", 0f);
```

Comment:
Is this right? what does this do?

Reply:
If It means that if the json's metallicSmoothnessTexturePath (i.e. the map) exists, apply it to the Metallic_GlossMap channel (the under-the-hood name for MetallicSmoothness). If it doesn't, set the material's Glossiness (i.e. Smoothness) to 0. I suppose I should also add a "Set Metallic to 0 if no MetallicSmoothness map exists" line as well, though.